libcontainer/nsenter/nsexec.c

```diff
@@ -97,13 +109,67 @@ void nsexec()
 		exit(1);
 	}
 
-	console = getenv("_LIBCONTAINER_CONSOLE_PATH");
-	if (console != NULL) {
-		consolefd = open(console, O_RDWR);
-		if (consolefd < 0) {
-			pr_perror("Failed to open console %s", console);
+	static char nlbuf[16384];
+	struct iovec iov = { nlbuf, sizeof(nlbuf) };
+	struct msghdr msg;
+	struct nlmsghdr *nh;
+
+	memset(&msg, 0, sizeof(msg));
+	msg.msg_name = &nh;
+	msg.msg_namelen = sizeof(nh);
+	msg.msg_iov = &iov;
+	msg.msg_iovlen = 1;
+	while (1) {
+		len = recvmsg(pipenum, &msg, 0);
```

Comment:
You need to read a message header, get the message len from it and read a message body. Otherwise you can read a part of foreing data or read incomplete message